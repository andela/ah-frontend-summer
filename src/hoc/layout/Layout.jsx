/* eslint-disable react/destructuring-assignment */
import  React, { Component } from 'react';
import propTypes from 'prop-types';
import Navbar from '../../components/Navbar';
import Footer from '../../components/Footer';

class Layout extends Component {
    render() {
        const { children } = this.props;
        return (
            <div>
                <Navbar />
                <div className="content-wrapper">
                    {children}
                </div>
                <Footer />
            </div>
        );
    }
}

Layout.propTypes = {
    children: propTypes.node.isRequired
};

export default Layout;
