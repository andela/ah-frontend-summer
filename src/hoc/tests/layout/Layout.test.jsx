import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../../layout/Layout';

describe('Layout tests', () => {
    const wrapper = shallow(<Layout />);
    test('render component without failing', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
