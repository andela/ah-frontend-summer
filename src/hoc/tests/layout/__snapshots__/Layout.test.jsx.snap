// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests render component without failing 1`] = `
<div>
  <Connect(Navbar) />
  <div
    className="content-wrapper"
  />
  <Footer />
</div>
`;
